### PR TITLE
feat(go): add temporary package name if missing

### LINKED ISSUE #151: golang formatter: problem when code snippet is not a full file
````markdown
```go
var err = errors.New("dddd")

type Whatever struct {
     StartAt                time.Time  `json:"start_at"`
                End                time.Time  `json:"end_at"`
    Delete bool `json:-"`
}
```
````

golang can only be formatted with tools if the file is valid.

So if the linting doesn't work you can try to add this to the begining of the file

````markdown
```go
package main

// REMOVE ANYTHING ABOVE THIS

var err = errors.New("dddd")

type Whatever struct {
     StartAt                time.Time  `json:"start_at"`
                End                time.Time  `json:"end_at"`
    Delete bool `json:-"`
}
```
````

mdsf will then be able to format like this

````markdown
```go
package main

// REMOVE ANYTHING ABOVE THIS

var err = errors.New("dddd")

type Whatever struct {
	StartAt time.Time `json:"start_at"`
	End     time.Time `json:"end_at"`
	Delete  bool      `json:-"`
}
```
````

Then if you trim the code then

````markdown
```go
var err = errors.New("dddd")

type Whatever struct {
	StartAt time.Time `json:"start_at"`
	End     time.Time `json:"end_at"`
	Delete  bool      `json:-"`
}
```
````

It would be a more fail proof hack but quite an hack

### PR DESCRIPTION
**Pull request by hougesen**
Closes #151

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -53,11 +55,21 @@ fn format_file(config: &MdsfConfig, input: &str) -> (bool, String) {
                 if is_snippet {
                     print_line_info(language, line_index + 1, line_index + snippet_lines + 1);
 
+                    if language == Language::Go && !code_snippet.contains("package ") {
```
Comment (ccoVeille) [inline src/lib.rs:58]: This could be a source of bug

````markdown
```go

// my package is rich
var err string
```
````

I would say the good way to test is the fact there is one line starting with "\s*package "

Please note, it's not necessarily the first as you can get comments in front of the package declaration, and the space in frot because copy pasted code could be misaligned with tab or spaces

````markdown
```go
    // foo provides teapot
    // yeah
    package foo
  
            // foo bad idented
    type foo    string
  type bar               int
```
```

**Review by ccoVeille — CHANGES_REQUESTED** [review]

File: src/lib.rs (hunk 1 above)
Comment (ccoVeille) [reply src/lib.rs:58]: That's why I talked about detecting

^\s*package\s+\w

But it will catch this

````markdown
```go

/* foo

    package foo

*/

type   foo    string

```
````

I think you should add it to an false negative edge case in the test. As a reminder for a later refactoring.

Something like: todo compromise

Here you are getting feedback only from me on this feature

Comment (hougesen) [reply src/lib.rs:58]: It is definitely helping a lot 😃 
Not a go user so didn't really notice until you mentioned it.

Comment (ccoVeille) [reply src/lib.rs:58]: I'm not a Rust developer 😅

But I'm very good at reverse engineering code 😜

I'm helping you because it will help you to build a great too I could use to format tons of documentation I have.

Comment (hougesen) [reply src/lib.rs:58]: I added the regex stuff in #159.   Gonna skip the multiline comment stuff for now. I am hoping it is an edge case that people wont hit 😅  Most likely gonna remove the regex stuff altogether and just check the snippet directly.

Comment (hougesen) [reply src/lib.rs:58]: I also added an issue so I don't forget :D  (#160)
